source/h2_decoder.c

```diff
@@ -1298,6 +1341,23 @@ static int s_state_fn_header_block_loop(struct aws_h2_decoder *decoder, struct a
                 return AWS_OP_ERR;
             }
 
+            if (decoder->header_block_in_progress.cookies.len) {
```

Comment:
don't flush the cookie if the header block is malformed. Once we detect the header block is malformed we stop delivering individual headers.

That behavior would be more clear if we also moved the `s_flush_pseudoheaders()` call inside the `if (!malformed)` block. I thought it was simpler-looking to have that function do its own check, but it's obviously more confusing then helpful